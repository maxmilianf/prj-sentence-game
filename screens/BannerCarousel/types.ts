import { ViewStyle } from 'react-native';

export type BannerCarouselStylesProps = {
  linearGradient: ViewStyle;
  bannerCarouselContainer: ViewStyle;
  carouselActiveDotStyle: ViewStyle;
  carouselStyle: ViewStyle;
  carouselPaginationStyle: ViewStyle;
};
