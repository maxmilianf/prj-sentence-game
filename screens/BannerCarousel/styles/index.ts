import { StyleSheet } from 'react-native';
import { BannerCarouselStyles } from '../types';

export const bannerCarouselStyles = StyleSheet.create<BannerCarouselStyles>({
  linearGradient: {
    flex: 1,
    height: '100%',
    minHeight: '100%',
  },
  bannerCarouselContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    minHeight: '100%',
  },
  carouselActiveDotStyle: {
    opacity: 0.8,
  },
  carouselStyle: {
    height: '60%',
  },
  carouselPaginationStyle: {
    marginBottom: 50,
  },
});
