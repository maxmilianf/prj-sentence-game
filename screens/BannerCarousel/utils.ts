export const gradientBackground = ['#a18cd1', '#fbc2eb'];
export const prevButtonName = 'chevron-left-box';
export const nextButtonName = 'chevron-right-box';
