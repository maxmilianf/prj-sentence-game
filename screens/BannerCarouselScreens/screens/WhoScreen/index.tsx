import React, { useContext } from 'react';
import { SentenceContext } from '../../../../context/SentenceContext';
import BannerCarouselItem from '../../../BannerCarousel/components/bannerCarouselItem';
import { ScreenTypes } from '../types';

const WhoScreen = ({ slideCarousel = () => {}, screenName }: ScreenTypes) => {
  const { whoSentence, setWhoSentence, whatInputRef, whoInputRef } =
    useContext(SentenceContext);

  return (
    <BannerCarouselItem
      capitalize
      screenName={screenName}
      partOfSentence={whoSentence}
      setPartOfSentence={setWhoSentence}
      slideCarousel={slideCarousel}
      inputRef={whoInputRef}
      inputRefToFocus={whatInputRef}
    />
  );
};

export default WhoScreen;
