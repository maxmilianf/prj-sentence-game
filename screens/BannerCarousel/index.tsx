import React, { MutableRefObject, useRef, useContext, useState } from 'react';
import {
  KeyboardAvoidingView,
  ScrollView,
  TextInput,
  View,
} from 'react-native';
import Swiper from 'react-native-swiper';
import { gradientBackground, prevButtonName, nextButtonName } from './utils';
import { LinearGradient } from 'expo-linear-gradient';
import { SentenceContext } from '../../context/SentenceContext';
import welcomeModal from './components/welcomeModal/index';
import bannerCarouselSkipButton from '../BannerCarousel/components/bannerCarouselSkipButtons';
import BannerCarouselScreens from '../BannerCarouselScreens';
import { bannerCarouselStyles } from './styles';

const BannerCarousel = () => {
  const styles = bannerCarouselStyles;
  const carouselRef = useRef<Swiper>(null);
  const {
    whoInputRef,
    setWhoSentence,
    setWhatSentence,
    setWhenSentence,
    setWhereSentence,
  } = useContext(SentenceContext);
  const [modalVisibility, setModalVisibility] = useState(true);

  const focusInput = (
    inputRef?: MutableRefObject<TextInput | null> | undefined | null
  ) => {
    if (inputRef) {
      setTimeout(() => {
        inputRef.current?.focus();
      }, 600);
    }
  };

  const slideCarousel = (
    inputRef?: MutableRefObject<TextInput> | undefined
  ) => {
    if (carouselRef) {
      carouselRef.current?.scrollBy(1, true);
      focusInput(inputRef);
    }
  };

  const slideCarouselToStart = () => {
    if (carouselRef) {
      setWhoSentence('');
      setWhatSentence('');
      setWhenSentence('');
      setWhereSentence('');
      carouselRef.current?.scrollTo(0, true);
      focusInput(whoInputRef);
    }
  };

  const onClosemodal = () => {
    setModalVisibility(false);
    if (whoInputRef) {
      setTimeout(() => {
        whoInputRef.current?.focus();
      }, 300);
    }
  };

  return (
    <ScrollView scrollEnabled={false}>
      <KeyboardAvoidingView>
        {welcomeModal({ modalVisibility, onClosemodal })}
        <LinearGradient
          colors={gradientBackground}
          style={styles.linearGradient}
        >
          <View style={styles.bannerCarouselContainer}>
            {/* 
        // @ts-ignore react-native-types version most likely*/}
            <Swiper
              ref={carouselRef}
              nextButton={bannerCarouselSkipButton({
                iconName: nextButtonName,
              })}
              prevButton={bannerCarouselSkipButton({
                iconName: prevButtonName,
              })}
              activeDotColor={'black'}
              activeDotStyle={styles.carouselActiveDotStyle}
              loop={false}
              horizontal
              showsButtons
              style={styles.carouselStyle}
              paginationStyle={styles.carouselPaginationStyle}
            >
              {BannerCarouselScreens({ slideCarousel, slideCarouselToStart })}
            </Swiper>
          </View>
        </LinearGradient>
      </KeyboardAvoidingView>
    </ScrollView>
  );
};

export default BannerCarousel;
