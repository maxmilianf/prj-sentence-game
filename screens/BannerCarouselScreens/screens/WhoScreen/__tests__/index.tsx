import React from 'react';
import renderer from 'react-test-renderer';
import { screenNames } from '../../../../BannerCarousel/utils';

import WhoScreen from '../index';

it('WhoScreen should render correctly', () => {
  const tree = renderer
    .create(
      <WhoScreen
        slideCarousel={() => {}}
        screenName={screenNames.who}
      ></WhoScreen>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
