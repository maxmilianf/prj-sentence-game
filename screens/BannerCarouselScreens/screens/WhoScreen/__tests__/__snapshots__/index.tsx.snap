// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WhoScreen should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginTop": 150,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000000",
            "fontFamily": "System",
            "fontWeight": "400",
          },
          Object {
            "textAlign": "left",
          },
          Array [
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "fontFamily": "System",
              "fontWeight": "400",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "fontSize": 24,
                "letterSpacing": 0,
                "lineHeight": 32,
                "marginVertical": 2,
              },
              Object {
                "opacity": 0.8,
              },
            ],
          ],
        ]
      }
    >
      Who?
    </Text>
  </View>
  <View
    style={
      Object {
        "marginTop": 20,
        "width": "75%",
      }
    }
  >
    <View
      pointerEvents="none"
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 6,
          },
          Object {
            "backgroundColor": "transparent",
            "borderColor": "rgba(0, 0, 0, 0.54)",
            "borderRadius": 4,
            "borderWidth": 1,
          },
        ]
      }
      testID="text-input-outline"
    />
    <View>
      <View
        style={
          Array [
            Object {
              "paddingBottom": 0,
            },
            Object {
              "minHeight": 40,
              "paddingTop": 8,
            },
          ]
        }
      >
        <TextInput
          autoCapitalize="sentences"
          editable={true}
          maxFontSizeMultiplier={1.5}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          onSubmitEditing={[Function]}
          placeholderTextColor="rgba(0, 0, 0, 0.54)"
          selectionColor="rgba(0, 0, 0, 0.8)"
          style={
            Array [
              Object {
                "flexGrow": 1,
                "margin": 0,
                "paddingHorizontal": 14,
                "zIndex": 1,
              },
              Object {
                "height": 40,
              },
              Object {
                "paddingBottom": 12,
                "paddingTop": 12,
              },
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": undefined,
                "textAlign": "left",
                "textAlignVertical": "center",
              },
              false,
              Array [
                Object {},
              ],
            ]
          }
          testID="text-input-outlined"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
</View>
`;
